---
layout: page
title: "FHIR Chat  · Internal error, when executing the following query on HAPI · implementers"
---

{% raw %}<h2>Stream: <a href="https://fhir.github.io/chat-archive/stream/179166-implementers/index.html">implementers</a></h2>
<h3>Topic: <a href="https://fhir.github.io/chat-archive/stream/179166-implementers/topic/Internal.20error.2C.20when.20executing.20the.20following.20query.20on.20HAPI.html">Internal error, when executing the following query on HAPI</a></h3>

<hr>

<base href="https://chat.fhir.org">

<head><link href="https://fhir.github.io/chat-archive/style.css" rel="stylesheet"></head>

<a name="153920560"></a>
<h4><a href="https://chat.fhir.org#narrow/stream/179166-implementers/topic/Internal%20error%2C%20when%20executing%20the%20following%20query%20on%20HAPI/near/153920560" class="zl"><img src="https://fhir.github.io/chat-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Dimitar <a href="https://fhir.github.io/chat-archive/stream/179166-implementers/topic/Internal.20error.2C.20when.20executing.20the.20following.20query.20on.20HAPI.html#153920560">(Nov 21 2017 at 09:29)</a>:</h4>
<p>Hello,<br>
I am getting the following internal error, whenever I am executing the following query on the HAPI server "<a href="http://fhirtest.uhn.ca/baseDstu3/Encounter?patient=330828&amp;date=ge2017-01-01&amp;_include:recurse=Encounter:practitioner&amp;_lastUpdated=ge2017-11-10" target="_blank" title="http://fhirtest.uhn.ca/baseDstu3/Encounter?patient=330828&amp;date=ge2017-01-01&amp;_include:recurse=Encounter:practitioner&amp;_lastUpdated=ge2017-11-10">http://fhirtest.uhn.ca/baseDstu3/Encounter?patient=330828&amp;date=ge2017-01-01&amp;_include:recurse=Encounter:practitioner&amp;_lastUpdated=ge2017-11-10</a>" </p>
<p><a href="/user_uploads/10155/YxUrkUuzjRZvyCPTI0de1sCR/pasted_image.png" target="_blank" title="pasted_image.png">pasted image</a> <br>
The strange thing is, whenever I omit the _lastUpdated parameter, this search query executes properly. Is it an issue on the HAPI server?</p>
<div class="message_inline_image"><a href="/user_uploads/10155/YxUrkUuzjRZvyCPTI0de1sCR/pasted_image.png" target="_blank" title="pasted image"><img src="/user_uploads/10155/YxUrkUuzjRZvyCPTI0de1sCR/pasted_image.png"></a></div>



<a name="153920568"></a>
<h4><a href="https://chat.fhir.org#narrow/stream/179166-implementers/topic/Internal%20error%2C%20when%20executing%20the%20following%20query%20on%20HAPI/near/153920568" class="zl"><img src="https://fhir.github.io/chat-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Dimitar <a href="https://fhir.github.io/chat-archive/stream/179166-implementers/topic/Internal.20error.2C.20when.20executing.20the.20following.20query.20on.20HAPI.html#153920568">(Nov 21 2017 at 09:39)</a>:</h4>
<p>Can a _lastUpdated parameter, not be used together with a recurse include or revInclude?</p>



<a name="153920571"></a>
<h4><a href="https://chat.fhir.org#narrow/stream/179166-implementers/topic/Internal%20error%2C%20when%20executing%20the%20following%20query%20on%20HAPI/near/153920571" class="zl"><img src="https://fhir.github.io/chat-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Grahame Grieve <a href="https://fhir.github.io/chat-archive/stream/179166-implementers/topic/Internal.20error.2C.20when.20executing.20the.20following.20query.20on.20HAPI.html#153920571">(Nov 21 2017 at 09:41)</a>:</h4>
<p>it can't, but it doesn't apply to the included stuff</p>



<a name="153920575"></a>
<h4><a href="https://chat.fhir.org#narrow/stream/179166-implementers/topic/Internal%20error%2C%20when%20executing%20the%20following%20query%20on%20HAPI/near/153920575" class="zl"><img src="https://fhir.github.io/chat-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Dimitar <a href="https://fhir.github.io/chat-archive/stream/179166-implementers/topic/Internal.20error.2C.20when.20executing.20the.20following.20query.20on.20HAPI.html#153920575">(Nov 21 2017 at 09:46)</a>:</h4>
<p>What is the meaning of the error in that case?</p>



<a name="153920576"></a>
<h4><a href="https://chat.fhir.org#narrow/stream/179166-implementers/topic/Internal%20error%2C%20when%20executing%20the%20following%20query%20on%20HAPI/near/153920576" class="zl"><img src="https://fhir.github.io/chat-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Grahame Grieve <a href="https://fhir.github.io/chat-archive/stream/179166-implementers/topic/Internal.20error.2C.20when.20executing.20the.20following.20query.20on.20HAPI.html#153920576">(Nov 21 2017 at 09:47)</a>:</h4>
<p>don't know - maybe ask on the HAPI email list?</p>



<a name="153920577"></a>
<h4><a href="https://chat.fhir.org#narrow/stream/179166-implementers/topic/Internal%20error%2C%20when%20executing%20the%20following%20query%20on%20HAPI/near/153920577" class="zl"><img src="https://fhir.github.io/chat-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Dimitar <a href="https://fhir.github.io/chat-archive/stream/179166-implementers/topic/Internal.20error.2C.20when.20executing.20the.20following.20query.20on.20HAPI.html#153920577">(Nov 21 2017 at 09:47)</a>:</h4>
<p><span class="user-mention" data-user-id="191316">@Grahame Grieve</span>  alright, will do. But technically, this query should execute, right?</p>



<a name="153920578"></a>
<h4><a href="https://chat.fhir.org#narrow/stream/179166-implementers/topic/Internal%20error%2C%20when%20executing%20the%20following%20query%20on%20HAPI/near/153920578" class="zl"><img src="https://fhir.github.io/chat-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Grahame Grieve <a href="https://fhir.github.io/chat-archive/stream/179166-implementers/topic/Internal.20error.2C.20when.20executing.20the.20following.20query.20on.20HAPI.html#153920578">(Nov 21 2017 at 09:48)</a>:</h4>
<p>y</p>



<a name="153921525"></a>
<h4><a href="https://chat.fhir.org#narrow/stream/179166-implementers/topic/Internal%20error%2C%20when%20executing%20the%20following%20query%20on%20HAPI/near/153921525" class="zl"><img src="https://fhir.github.io/chat-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> James Agnew <a href="https://fhir.github.io/chat-archive/stream/179166-implementers/topic/Internal.20error.2C.20when.20executing.20the.20following.20query.20on.20HAPI.html#153921525">(Nov 24 2017 at 18:32)</a>:</h4>
<p>Hi <span class="user-mention" data-user-id="194501">@Dimitar</span> , this was a bug on the server. It has been fixed, and the fix is in the current 3.2.0 SNAPSHOT build. :)</p>



<a name="153921560"></a>
<h4><a href="https://chat.fhir.org#narrow/stream/179166-implementers/topic/Internal%20error%2C%20when%20executing%20the%20following%20query%20on%20HAPI/near/153921560" class="zl"><img src="https://fhir.github.io/chat-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> Dimitar <a href="https://fhir.github.io/chat-archive/stream/179166-implementers/topic/Internal.20error.2C.20when.20executing.20the.20following.20query.20on.20HAPI.html#153921560">(Nov 24 2017 at 23:53)</a>:</h4>
<p><span class="user-mention" data-user-id="191319">@James Agnew</span>  Great! Glad I could help :), if I did anyways. I implemented a workaround, just in case, I get it elsewhere, removing the  _lastUpdated search parameter, from the search params, if I catch an error like that. What was the problem with it anyways?</p>



<a name="153921564"></a>
<h4><a href="https://chat.fhir.org#narrow/stream/179166-implementers/topic/Internal%20error%2C%20when%20executing%20the%20following%20query%20on%20HAPI/near/153921564" class="zl"><img src="https://fhir.github.io/chat-archive/assets/img/zulip.svg" alt="view this post on Zulip" style="width:20px;height:20px;"></a> James Agnew <a href="https://fhir.github.io/chat-archive/stream/179166-implementers/topic/Internal.20error.2C.20when.20executing.20the.20following.20query.20on.20HAPI.html#153921564">(Nov 25 2017 at 02:37)</a>:</h4>
<p>Essentially it was a bug with the _include handling. Oddly enough it was more to do with the recursive _include where the second round of including failed if _lastUpdate left it with nothing to include. Blah!</p>


{% endraw %}
<hr><p>Last updated: Apr 12 2022 at 19:14 UTC</p>
